step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageUpdateParameterBuilder.java

```diff
@@ -141,17 +141,28 @@ public AutomationPackageUpdateParameterBuilder forRedeployPackage(AutomationPack
         }
         this.automationPackageVersion = oldPackage.getVersion();
         this.activationExpression = oldPackage.getActivationExpression() != null ? oldPackage.getActivationExpression().getScript() : null;
-        this.enricher = parentParameters.enricher;
-        this.objectPredicate = parentParameters.objectPredicate;
-        this.writeAccessPredicate = parentParameters.writeAccessPredicate;
+
+        //We need to rebuild the context to get the proper object enricher
+        //For redeployment of linked package we assume that if they reference the resource that got updated, we are allowed to refresh them
+        AbstractContext context = new AbstractContext() { };
+        try {
+            objectHookRegistry.rebuildContext(context, oldPackage);
+        } catch (Exception e) {
+            String errorMessage = "Error while rebuilding context for origin automation package " + AutomationPackageManager.getLogRepresentation(oldPackage);
+            throw new RuntimeException(errorMessage, e);
+        }
+        this.enricher = objectHookRegistry.getObjectEnricher(context);
+        this.objectPredicate =  o -> true;
```

Comment:
This code is explicitly disabling some security features - add a comment why this is safe (i.e. "only used temporarily, required because..." or whatever.